docs/rules/jsx-max-props-per-line.md

```diff
@@ -56,6 +60,28 @@ The following patterns are not considered warnings:
 />;
 ```
 
+### `when`
+
+Possible values:
+- `always` (default) - Always check for max props per line.
+- `multiline` - Only check for max props per line when jsx tag spans multiple lines.
+
+The following patterns are considered warnings:
+```jsx
+// [1, {when: always}]
```

Comment:
I think json format is the most common by a massive amount, so that's the format we should stick to in docs.